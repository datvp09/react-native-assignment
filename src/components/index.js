import TripInfoItem from './TripInfoItem';

export {TripInfoItem};
