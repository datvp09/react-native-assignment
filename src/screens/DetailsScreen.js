import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {connect} from 'react-redux';
import moment from 'moment';

export class DetailsScreen extends React.Component {
  static navigationOptions = {
    title: 'Details',
  };

  render() {
    /* 2. Get the param, provide a fallback value if not available */
    const {navigation} = this.props;
    const item = navigation.getParam('item', {});
    const {trip_id, distance, duration, end_time, start_time} = item;
    const startTimeText = moment(start_time)
      .utc()
      .format('H:mm, D/M/YYYY');
    const endTimeText = moment(end_time)
      .utc()
      .format('H:mm, D/M/YYYY');

    return (
      <View style={styles.container}>
        <Text style={styles.text}>
          {'TripId: '}
          <Text style={styles.textRegular}>{trip_id}</Text>
        </Text>
        <Text style={styles.text}>
          {'Distance: '}
          <Text style={styles.textRegular}>{distance}</Text>
        </Text>
        <Text style={styles.text}>
          {'Duration: '}
          <Text style={styles.textRegular}>{duration}</Text>
        </Text>
        <Text style={styles.text}>
          {'Start time: '}
          <Text style={styles.textRegular}>{startTimeText}</Text>
        </Text>
        <Text style={styles.text}>
          {'End time: '}
          <Text style={styles.textRegular}>{endTimeText}</Text>
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 15,
  },
  text: {
    color: 'black',
    fontSize: 16,
    fontWeight: '600',
    marginBottom: 5,
  },
  textRegular: {
    fontWeight: 'normal',
  },
});

const mapStateToProps = state => {
  return {
    trips: state.trip.trips,
  };
};

export default connect(mapStateToProps)(DetailsScreen);
