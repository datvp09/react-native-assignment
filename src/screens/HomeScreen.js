import React from "react"
import { StyleSheet, View, FlatList } from "react-native"
import axios from "axios"
import { connect } from "react-redux"
import { TripInfoItem } from "../components"

export class HomeScreen extends React.Component {
  static navigationOptions = {
    title: "Home"
  }

  state = {
    trips: [],
    isLoading: false
  }

  componentDidMount() {
    const { saveTrips } = this.props

    this.onRefresh()
  }

  onRefresh = () => {
    axios("https://api.gigacover.com/recruitment/challenge")
      .then(res => {
        if (!res || res.status != 200) {
          Toast("Lỗi dữ liệu trả về")
          return
        }
        this.setState({ trips: res.data.trips })
      })
      .catch(e => {
        Toast("Lỗi dữ liệu trả về")
        return
      })
  }

  onItemPress = (item, index) => {
    this.props.navigation.navigate("Details", {
      item,
      index
    })
  }

  renderItems = ({ item, index }) => {
    const props = {
      item: item,
      onPress: () => this.onItemPress(item, index)
    }
    if (index == 0) {
      props.style = { marginTop: 10 }
    }

    return <TripInfoItem {...props} />
  }

  render() {
    const { trips, isLoading } = this.state

    return (
      <FlatList
        data={trips}
        refreshing={isLoading}
        onRefresh={this.onRefresh}
        renderItem={this.renderItems}
        keyExtractor={item => item.trip_id.toString()}
        style={styles.list}
        contentContainerStyle={styles.contentContainer}
      />
    )
  }
}

const styles = StyleSheet.create({
  contentContainer: {
    flexGrow: 1
  },
  list: {
    flex: 1,
    backgroundColor: "#ddd",
    borderWidth: 1,
    borderColor: "red"
  }
})

const mapDispatchToProps = dispatch => {
  const { saveTrips } = require("../../redux/TripReducer")

  return {
    saveTrips: () => dispatch(saveTrips())
  }
}

export default connect(null, mapDispatchToProps)(HomeScreen)
