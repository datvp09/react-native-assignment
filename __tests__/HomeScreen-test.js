import "react-native"
import React from "react"
import { HomeScreen } from "../src/screens"
import renderer from "react-test-renderer"
import { Provider } from "react-redux"
import configureMockStore from "redux-mock-store"

const mockStore = configureMockStore()
const store = mockStore({})

describe("HomeScreen", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const component = renderer
        .create(
          <Provider store={store}>
            <HomeScreen />
          </Provider>
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
