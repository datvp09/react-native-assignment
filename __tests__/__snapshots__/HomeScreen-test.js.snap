// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Rendering should match to snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onRefresh={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  refreshing={false}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#ddd",
      "borderColor": "red",
      "borderWidth": 1,
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTRefreshControl />
  <View />
</RCTScrollView>
`;
